### PR TITLE
[MARVP-35] More menu form

### PR DESCRIPTION
**Pull request by mishradileep**
Related to : https://konysolutions.atlassian.net/browse/MARVP-35

### CODE REVIEW

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        kony.ui.createAnimation({
+          0:{left:"-80%","stepConfig":{}},
+          100:{left:"10%","stepConfig":{}}}),
+        {delay:delayThrashold,fillMode:kony.anim.FILL_MODE_FORWARDS,duration:self.animationDuration},
+        {animationEnd: function() {}}
+      );
+      delayThrashold = delayThrashold + 0.2;
+    });
+  },
+  /** @function menuAnimateSlideOut
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Please change the function comments style. We are using the below style for the complete application.

 /**
     * @function getCoordinates
     * @descripiton This function is used to construct the coOrdinates Information
     * @param eventLocationData The complete information about the event
     * @private
     */

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  },
+  
+  menuBindOnClick: function() {
+    var self = this;
+    let menuWidgets = this.view.flxMoreWrapper.widgets();
+    menuWidgets.forEach(function(currentMenu, index){
+      currentMenu.onClick = self.menuBindOnClickAction.bind(self);
+    });
+  },
+
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Please add function comments wherever applicable

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      currentMenu.onClick = self.menuBindOnClickAction.bind(self);
+    });
+  },
+  
+  menuBindOnClickAction: function(eventObj){
+    var self = this;
+    self.menuAnimateSlideOut(eventObj);
+    
+  }
+ });
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Please add a new line at the EOF

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
+define({ 
+
+ //Type your controller code here
```
Comment (vijaysamanthapudi) [inline KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Remove commented text

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  animationDuration : 0.5,
+  menuAnimateSlideIn: function(){
+    var self = this;
+    let menuWidgets = this.view.flxMoreWrapper.widgets();
+    let delayThrashold = 0;
+    menuWidgets.forEach(function(currentMenu, index){
+      let currentItemIdex = index;
+      currentMenu.animate(
+        kony.ui.createAnimation({
+          0:{left:"-80%","stepConfig":{}},
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Can't we set the default value of left to "-80%" and just have the step for 100 here?

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        {delay:delayThrashold,fillMode:kony.anim.FILL_MODE_FORWARDS,duration:self.animationDuration},
+        {animationEnd: function() {}}
+      );
+      delayThrashold = delayThrashold + 0.2;
+    });
+  },
+  /** @function menuAnimateSlideOut
+   *  		    Slide Out menu items one by one
+   */
+  menuAnimateSlideOut: function(eventObj){
```
Comment (krishk95) [inline KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Can we merge the two animation functions and execute the functionality based on a flag/param?

**Review by krishk95 — CHANGES_REQUESTED** [review]
Please address the PR Comments

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js (hunk 1 above)
Comment (mishradileep) [reply KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Done

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js (hunk 4 above)
Comment (mishradileep) [reply KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Done

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js (hunk 5 above)
Comment (mishradileep) [reply KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Done

File: KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js (hunk 6 above)
Comment (mishradileep) [reply KonyAccelerate/controllers/mobile/frmMoreController/frmMoreController.js]: Done

**Review by krishk95 — APPROVED** [review]

**Review by vigneshwarankarunanithi — APPROVED** [review]

**Pull request merged**